Cargo check on push and pull request (github actions)

vvilhonen opened the pull request:
Tests require bit more work to run on CI but checking compilation is better than nothing

1c3t3a commented on .github/workflows/ci.yml:
```diff
@@ -0,0 +1,20 @@
+      - uses: dtolnay/rust-toolchain@master
+        with:
+          toolchain: stable
+      - run: cargo check
```
maybe add `--all-features`?

vvilhonen replied:
Good idea!

1c3t3a commented on .github/workflows/ci.yml:
```diff
@@ -0,0 +1,20 @@
+name: CI
+
+on:
+  push:
```
Maybe restrict this only to the `master` branch?

vvilhonen replied:
Makes sense, I’ll update this along with the previous comment’s change

vvilhonen replied:
I tried with it and the build wouldn't trigger anymore in my fork's branch after push so there's no way to test changes to gh actions configuration.

mdrokz reviewed (approved):
Looks good thanks for your contribution!

vvilhonen commented:
- Cargo check is now done with `--all-features`
- Warnings turned into errors with `RUSTFLAGS=-Dwarnings`
- Fixed everything the previous changes brought up
- Added format check

mdrokz commented:
> * Cargo check is now done with `--all-features`
> * Warnings turned into errors with `RUSTFLAGS=-Dwarnings`
> * Fixed everything the previous changes brought up
> * Added format check
> * Build status badge pointing to github actions

great job and thanks for your contribution!

mdrokz commented:
> * Cargo check is now done with `--all-features`
> * Warnings turned into errors with `RUSTFLAGS=-Dwarnings`
> * Fixed everything the previous changes brought up
> * Added format check
> * Build status badge pointing to github actions

great job and thanks for your contribution!
